src/Sylius/Behat/Context/Api/Shop/AddressBookContext.php

```diff
@@ -0,0 +1,225 @@
+use Sylius\Component\Core\Model\AddressInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Repository\AddressRepositoryInterface;
+use Webmozart\Assert\Assert;
+
+final class AddressBookContext implements Context
+{
+    /** @var ApiClientInterface */
+    private $client;
+
+    /** @var ResponseCheckerInterface */
+    private $responseChecker;
+
+    /** @var AddressRepositoryInterface */
+    private $addressRepository;
+
+    /** @var IriConverterInterface */
+    private $iriConverter;
+
+    public function __construct(
+        ApiClientInterface $client,
+        ResponseCheckerInterface $responseChecker,
+        AddressRepositoryInterface $addressRepository,
+        IriConverterInterface $iriConverter
+    ) {
+        $this->client = $client;
+        $this->responseChecker = $responseChecker;
+        $this->addressRepository = $addressRepository;
+        $this->iriConverter = $iriConverter;
+    }
+
+    /**
+     * @Given I am editing the address of :fullName
+     */
+    public function iAmEditingTheAddressOf(string $fullName): void
+    {
+        $address = $this->getAddressOf($fullName);
+
+        $this->client->buildUpdateRequest((string) $address->getId());
+    }
+
+    /**
+     * @When I want to add a new address to my address book
+     */
+    public function iWantToAddANewAddressToMyAddressBook(): void
+    {
+        $this->client->buildCreateRequest();
+    }
+
+    /**
+     * @When /^I specify the (address as "([^"]+)", "([^"]+)", "([^"]+)", "([^"]+)", "([^"]+)", "([^"]+)")$/
+     */
+    public function iSpecifyTheAddressAs(AddressInterface $address): void
+    {
+        $this->client->setContent(
+            [
+                'countryCode' => $address->getCountryCode(),
+                'street' => $address->getStreet(),
+                'city' => $address->getCity(),
+                'postcode' => $address->getPostcode(),
+                'provinceName' => $address->getProvinceName(),
+            ]
+        );
+    }
+
+    /**
+     * @When I add it
+     */
+    public function iAddIt(): void
+    {
+        $this->client->create();
+    }
+
+    /**
+     * @When I leave every field empty
+     */
+    public function iLeaveEveryFieldEmpty(): void
+    {
+        $this->client->setContent([]);
+    }
+
+    /**
+     * @When I choose :countryName as my country
+     */
+    public function iChooseAsMyCountry(string $countryName): void
+    {
+        $this->client->addRequestData('country', $countryName);
+    }
+
+    /**
+     * @When I do not specify province
+     */
+    public function iDoNotSpecifyProvince(): void
+    {
+        $this->client->addRequestData('provinceName', null);
+        $this->client->addRequestData('provinceCode', null);
+    }
+
+    /**
+     * @When I remove the street
+     */
+    public function iRemoveTheStreet(): void
+    {
+        $this->client->addRequestData('street', null);
```

Comment:
Even if it works... shouldn't we literally remove this data from the request's content, not only set it as `null`?

Reply:
unfortunately i couldnt find a way to remove single parameter :/ setting it as null works well.

Reply:
Because probably, for now, we don't have the possibility to remove data, IMHO you can add new method to `client` sth like `removeRequestData`

Reply:
maybe we should add a new method to client? Or refactor it somehow